webapp/packages/plugin-user-profile/src/UserProfileForm/UserAuthenticationPart/ChangePassword.tsx

```diff
@@ -27,47 +63,110 @@ export const ChangePassword = observer<Props>(function ChangePassword({ state, d
+    async onSubmit() {
+      try {
+        await userInfoResource.updateLocalPassword(state.oldPassword, state.password);
+        resetForm();
+        notificationService.logSuccess({ title: 'plugin_user_profile_authentication_change_password_success' });
+      } catch (exception) {
+        if (exception instanceof Error) {
+          notificationService.logException(exception);
+        }
+      }
+    },
+  });
+
+  function resetForm() {
+    state.oldPassword = '';
+    state.password = '';
+    state.repeatedPassword = '';
+    form.ref?.reset();
+  }
+
+  function onChange(value: FormChangeValues, name: string | undefined) {
+    state[name as keyof IUserProfileFormAuthenticationState] = value as string;
+  }
+
+  useExecutor({
+    executor: userProfileOptionsPanelService.onClose,
+    handlers: [
+      async function closeHandler(_, contexts) {
+        const context = contexts.getContext(userProfileContext);
+
+        if (!isObjectsEqual(state, INITIAL_STATE) && !context.force) {
+          const result = await commonDialogService.open(ConfirmationDialog, {
+            title: 'plugin_user_profile_authentication_change_password_cancel_title',
+            message: 'plugin_user_profile_authentication_change_password_cancel_message',
+            confirmActionText: 'ui_processing_ok',
+          });
+
+          if (result === DialogueStateResult.Rejected) {
+            ExecutorInterrupter.interrupt(contexts);
+          }
+        }
+      },
+    ],
+  });
+
   return (
-    <Form>
-      <Group form gap>
-        <GroupTitle>{translate('plugin_user_profile_authentication_change_password')}</GroupTitle>
-        <InputField
-          type="password"
-          name="oldPassword"
-          state={state}
-          readOnly={disabled}
-          mapValue={(value?: string) => value?.trim() ?? ''}
-          small
-          required
-        >
-          {translate('plugin_user_profile_authentication_change_password_current_password')}
-        </InputField>
-        <InputField
-          ref={passwordValidationRef}
-          type="password"
-          name="password"
-          autoComplete="new-password"
-          state={state}
-          readOnly={disabled}
-          mapValue={(value?: string) => value?.trim() ?? ''}
-          small
-          required
-        >
-          {translate('plugin_user_profile_authentication_change_password_new_password')}
-        </InputField>
-        <InputField
-          ref={passwordRepeatRef}
-          type="password"
-          name="repeatedPassword"
-          state={state}
-          readOnly={disabled}
-          mapValue={(value?: string) => value?.trim() ?? ''}
-          small
-          required
-        >
-          {translate('plugin_user_profile_authentication_change_password_repeat_password')}
-        </InputField>
-      </Group>
-    </Form>
+    <ColoredContainer wrap overflow gap>
+      <Container medium gap>
+        <Form context={form} onChange={onChange}>
```

Comment:
You don't need this onChange. When you pass the `state` + `name` property to our form fields, the filed will mutate state:
`state[name] = value` on change